ruby/vscode/minitest/reporter.rb

```diff
@@ -63,7 +63,7 @@ def vscode_data
       end
 
       def vscode_result(r)
-        base = VSCode::Minitest.tests.find_by(klass: r.klass, method: r.name).deep_dup
+        base = VSCode::Minitest.tests.find_by(klass: r.klass, method: r.name).dup
```

Comment:
Based on [this blog post](http://sachinchoudhary.blogspot.com/2014/01/ruby-clone-dup-vs-rails-deepdup.html), the main difference is that any nested hashes will essentially be passed by reference instead of value, so if you change a nested hash within `base` it'll change the original Minitest hash too.

The only problem changing this to `dup` might cause would be the line where `base[:exception]`'s value is set to a nested hash. I'm not entirely sure if that'll actually cause any problems, though.

Reply:
cc: @cristianbica in case you have anything to add about why `deep_dup` was used here.

It seems to me that everything works fine, so I'm going to merge this.

Reply:
Should be ok. `deep_dup` was in there as a safety in case we will add in the future some arrays/hashes under the test data hash.